### PR TITLE
Tg download enrollment history

### PR DESCRIPTION
**Pull request by gala377**

### CODE REVIEW

**Comment by gala377** [comment]
Added report download for attendance as requested

File: apsi-enrollment-fe/src/app/modules/main/components/attendance-pdf/attendance-pdf.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  attendance?: Attendance[] = null;
+  readonly fileName: string = 'my-attendance';
+  constructor(private attendanceService: AttendanceService, private appRef: ApplicationRef) { }
+
+  ngOnInit() {}
+
+  onDownloadPDF() {
+    const data = this.prepareData();
+    data.pipe(
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/components/attendance-pdf/attendance-pdf.component.ts:27]: can be merged with previous line (you would use `data` single time anyways):
```ts
 this.prepareData().pipe(...).subscribe(...);
```

File: apsi-enrollment-fe/src/app/modules/main/components/enrollments-pdf/enrollments-pdf.component.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<div>
+  <button class="btn btn-primary" type="button" (click)="onDownloadPDF()">
+    Download
+  </button>
+  <div id="htmlData" #htmlData style="visibility: hidden;">
+    <table id="pdfData">
+        <thead class="table">
+          <tr>
+            <th>id</th>
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/components/enrollments-pdf/enrollments-pdf.component.html:9]: Use ngx-translate. Capitalize those strings.

File: apsi-enrollment-fe/src/app/modules/main/components/enrollments-pdf/enrollments-pdf.component.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  events?: BasicEvent[] = null;
+  readonly fileName: string = 'my-enrollments';
+
+  constructor(private eventService: EventService, private appRef: ApplicationRef) { }
+
+  ngOnInit() {}
+
+  onDownloadPDF() {
+    const data = this.prepareData();
+    data.pipe(
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/components/enrollments-pdf/enrollments-pdf.component.ts:28]: Use:
```ts
 this.prepareData().pipe(...).subscribe(...);
```

File: apsi-enrollment-fe/src/app/modules/main/pages/all-events/all-events.component.ts [hunk 4]
```diff
@@ -19,22 +19,25 @@ export class AllEventsComponent implements OnInit {
...
     }
 
-    var request = { pageNumber: state.page.current - 1, pageSize: state.page.size, searchQuery: searchString };
-    var events = this.getEvents(request);
+    const request = {
+      pageNumber: state.page.current - 1,
+      pageSize: state.page.size,
+      searchQuery: searchString,
+    };
+    const events = this.getEvents(request);
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/pages/all-events/all-events.component.ts:40]: could be meged with next line:
```ts
this.getEvents(request).subscribe(...)
```

File: apsi-enrollment-fe/src/app/modules/main/pages/my-attendances/my-attendances.component.ts [hunk 5]
```diff
@@ -1,44 +1,47 @@
...
+        (prev, next) => `${prev}${next.property}=${next.value},`, '');
+      searchString = searchString.substring(0, searchString.length - 1);
     }
+
+    const request = {
+      pageNumber: state.page.current - 1,
+      pageSize: state.page.size,
+      searchQuery: searchString,
+    };
+    const events = this.attendanceService.getMyAttendancePage(request);
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/pages/my-attendances/my-attendances.component.ts:35]: could be merged with next line

File: apsi-enrollment-fe/src/app/modules/main/pages/my-enrolled-events/my-enrolled-events.component.ts [hunk 6]
```diff
@@ -34,17 +34,21 @@ export class MyEnrolledEventsComponent implements OnInit {
...
     }
 
-    var request = { pageNumber: state.page.current - 1, pageSize: state.page.size, searchQuery: searchString };
-    var events = this.getEvents(request);
+    const request = {
+      pageNumber: state.page.current - 1,
+      pageSize: state.page.size,
+      searchQuery: searchString,
+    };
+    const events = this.getEvents(request);
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/pages/my-enrolled-events/my-enrolled-events.component.ts:42]: could be merged with next line

File: apsi-enrollment-fe/src/app/modules/main/pages/my-organizer-events/my-organizer-events.component.ts [hunk 7]
```diff
@@ -20,22 +20,24 @@ export class MyOrganizerEventsComponent implements OnInit {
...
     }
 
-    var request = { pageNumber: state.page.current - 1, pageSize: state.page.size, searchQuery: searchString };
-    var events = this.getEvents(request);
+    const request = {
+      pageNumber: state.page.current - 1,
+      pageSize: state.page.size,
+      searchQuery: searchString ,
+    };
+    const events = this.getEvents(request);
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/pages/my-organizer-events/my-organizer-events.component.ts:40]: could be merged with next line

File: apsi-enrollment-fe/src/app/modules/main/pages/my-speaker-events/my-speaker-events.component.ts [hunk 8]
```diff
@@ -22,20 +22,23 @@ export class MySpeakerEventsComponent implements OnInit {
...
     }
 
-    var request = { pageNumber: state.page.current - 1, pageSize: state.page.size, searchQuery: searchString };
-    var events = this.getEvents(request);
+    const request = { 
+      pageNumber: state.page.current - 1,
+      pageSize: state.page.size,
+      searchQuery: searchString,
+    };
+    const events = this.getEvents(request);
```
Comment (Grushenko) [inline apsi-enrollment-fe/src/app/modules/main/pages/my-speaker-events/my-speaker-events.component.ts:41]: could be merged with next line

**Pull request merged**